bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -125,6 +127,16 @@ def run_inner_plan(scope, plan_name, targets, args = {})
       params = wrap_sensitive_parameters(params, closure.parameters)
     end
 
+    # This can be anything as long as it's unique
+    plan_instance_id = SecureRandom.uuid
+
+    # Add the plan invocation ID to the plan_stack for the PlanFuture the plan is
+    # running in so that we know the PlanFuture is running in a new plan
+    # invocation. This can be nil in test cases and when `wait()` isn't
+    # supported.
+    current_future = executor.get_current_future(fiber: Fiber.current)
+    current_future&.plan_stack&.unshift(plan_instance_id)
```

Comment:
I think we can safely remove the safe nav operators?

Reply:
@lucywyman Heh, apparently not.

Reply:
I mean, I'm just too lazy to update all the tests :P